python/hail/dataset.py

```diff
@@ -403,7 +405,8 @@ def annotate_global(self, path, annotation, annotation_type):
         annotation_type._typecheck(annotation)
 
         annotated = self._jvds.annotateGlobal(annotation_type._convert_to_j(annotation), annotation_type._jtype, path)
-        assert annotated.globalSignature().typeCheck(annotated.globalAnnotation()), 'error in java type checking'
+        if sys.version_info < (3,):
+            assert annotated.globalSignature().typeCheck(annotated.globalAnnotation()), 'error in java type checking'
```

Comment:
What's the specific failure that this generates with py3? We could move this check into Scala, if that would solve the problem.